lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -353,28 +385,59 @@ def _on_relation_changed(self, event):
 
         # TODO future PR, leader unit verifies shard was added to cluster (update-status hook)
 
-    def _on_relation_broken(self, event) -> None:
-        """Waits for the shard to be fully drained from the cluster."""
+    def pass_hook_checks(self, event):
+        """Runs the pre-hooks checks for ConfigServerRequirer, returns True if all pass."""
         if self.charm.is_role(Config.Role.REPLICATION):
-            self.unit.status = BlockedStatus("role replication does not support sharding")
+            self.charm.unit.status = BlockedStatus("role replication does not support sharding")
             logger.error("sharding interface not supported with config role=replication")
-            return
+            return False
 
         if not self.charm.is_role(Config.Role.SHARD):
+            logger.info("skipping %s is only be executed by shards", type(event))
+            return False
+
+        if not self.charm.db_initialised:
+            event.defer()
+            return False
+
+        return True
+
+    def _on_relation_broken(self, event: RelationBrokenEvent) -> None:
+        """Waits for the shard to be fully drained from the cluster."""
+        if not self.pass_hook_checks(event):
+            logger.info("Skipping relation joined event: hook checks re not passed")
+            return
+
+        # Only relation_deparated events can check if scaling down
+        departed_relation_id = event.relation.id
+        if not self.charm.has_departed_run(departed_relation_id):
             logger.info(
-                "skipping relation broken event ShardingProvider is only be executed by shards"
+                "Deferring, must wait for relation departed hook to decide if relation should be removed."
             )
+            event.defer()
             return
 
-        if not self.charm.db_initialised:
-            event.defer()
+        # check if were scaling down and add a log message
+        if self.charm.is_scaling_down(event.relation.id):
+            logger.info(
+                "Relation broken event occurring due to scale down, do not proceed to remove users."
```

Comment:
```suggestion
                "Relation broken event occurring due to scale down, do not proceed to add/remove shards."
```

Reply:
I can add this, but I don't think it should go here, relation broken is only for removal